### PR TITLE
feat: text widget

### PR DESCRIPTION
**Pull request by dzbo**
### Ticket ID

[DEV-11807](https://app.clickup.com/t/2645698/DEV-11807)

### CODE REVIEW

File: domains/grid/components/AddWidgetText.vue [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+onMounted(() => {
+  setTimeout(() => {
+    const input = document?.querySelector(
+      'lukso-input'
+    ) as unknown as HTMLElement
+
+    input?.shadowRoot?.querySelector('input')?.focus()
+  }, INPUT_FOCUS_DELAY)
+
+  Object.assign(inputValues, widgetData.value?.properties)
```
Comment (doubleppereira) [inline domains/grid/components/AddWidgetText.vue:137]: Wouldnt here the `cloneObject` make sense?

**Review by JeneaVranceanu — APPROVED** [review]

**Review by doubleppereira — APPROVED** [review]
LGTM just a few comments

File: domains/grid/components/AddWidgetText.vue (hunk 1 above)
Comment (dzbo) [reply domains/grid/components/AddWidgetText.vue:137]: It's different case. Here we are not making copy of an object but we want to set widget properties on `inputValues` reactive property while we are in edit mode.

File: domains/grid/components/AddWidgetText.vue [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+  handleCancel()
+}
+
+const handleCancel = () => {
+  clearWidgetData()
+  closeModal()
+}
+
+const handleTitleChange = async (customEvent: CustomEvent) => {
```
[the remark this answers is not in the document]
Comment (dzbo) [reply domains/grid/components/AddWidgetText.vue:58]: Thanks for the hint I will take look on this library!

[the remark this answers is not in the document]
Comment (dzbo) [reply domains/grid/components/AddWidgetText.vue:58]: I've played around with this library but it doesn't reduce the code and complicate things a little bit due to fact that we work with web components. Maybe some validation library would make sense but still it seem like overkill for such simple forms.

**Pull request merged**